core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20211115_v1_workspace_migration_history.xml

```diff
@@ -0,0 +1,71 @@
+    <changeSet logicalFilePath="dummy" author="ehigham" id="create_V1_WORKSPACE_MIGRATION_HISTORY">
+        <createTable tableName="V1_WORKSPACE_MIGRATION_HISTORY"
+                     remarks="Records the history and progress of v1 -> v2 workspace migration attempts">
+            <column name="id" type="BIGINT" autoIncrement="true">
+                <constraints primaryKey="true"/>
+            </column>
+            <column name="WORKSPACE_ID"
+                    type="UUID"
+                    remarks="ID of the WORKSPACE being migrated.">
+                <constraints nullable="false"/>
+            </column>
+            <column name="CREATED"
+                    type="DATETIME"
+                    defaultValueComputed="CURRENT_TIMESTAMP"
+                    remarks="When the system scheduled migrating the WORKSPACE.">
+                <constraints nullable="false"/>
+            </column>
+            <column name="STARTED"
+                    type="DATETIME"
+                    remarks="When the system started migrating the WORKSPACE.">
+                <constraints nullable="true"/>
+            </column>
+            <column name="FINISHED"
+                    type="DATETIME"
+                    remarks="When the system finished migrating the WORKSPACE.">
+                <constraints nullable="true"/>
+            </column>
+            <column name="OUTCOME"
+                    type="VARCHAR(16)"
+                    remarks="The state of migrating the WORKSPACE when the system FINISHED.">
+                <constraints nullable="true"/>
+            </column>
+            <column name="MESSAGE"
+                    type="TEXT"
+                    remarks="Any supplemental message generated by the system, including failure messages.">
+                <constraints nullable="true"/>
+            </column>
+        </createTable>
+    </changeSet>
+    <changeSet logicalFilePath="dummy" author="ehigham"
+               id="create_V1_WORKSPACE_MIGRATION_HISTORY_insertion_trigger">
+        <comment>
+            The task of migrating a v1 workspace to a v2 workspace might fail for a number of
+            reasons and it's important that we record these failures. An attempt at migrating any
+            particular workspace is represented by an insertion into V1_WORKSPACE_MIGRATION_HISTORY.
+            To prevent concurrent attempts, we'll only allow insertions when previous attempts have
+            FINISHED.
+        </comment>
+        <sql endDelimiter="~">
+            CREATE TRIGGER fail_insert_if_workspace_is_being_migrated
+            BEFORE INSERT ON V1_WORKSPACE_MIGRATION_HISTORY
+            FOR EACH ROW
+            BEGIN
+                IF (SELECT COUNT(*) FROM V1_WORKSPACE_MIGRATION_HISTORY h
+                    WHERE h.WORKSPACE_ID = NEW.WORKSPACE_ID AND h.FINISHED IS NULL)
+                THEN
+                    SET @error_message = CONCAT_WS(' ', 'Workspace', NEW.WORKSPACE_ID, 'is being migrated.');
```

Comment:
Where will this error message bubble up? Just in the logs?

Reply:
No, it'll be thrown on insert